doc/whats_new/v1.3.rst

```diff
@@ -294,6 +294,10 @@ Changelog
 :mod:`sklearn.linear_model`
 ...........................
 
+- |Efficiency| Avoid uncessary data scaling and conversion to sparse data
+  structures in :class:`linear_model.LinearRegression` when not necessary.
```

Comment:
Only for sparse? But we also have performance improvements for dense `X`, right?

Reply:
BTW, the conversion of dense to sparse came as a big surprise for me!

Reply:
Done.